<a href="https://colab.research.google.com/github/Lilian624/Microsoft-Movies-Project-1/blob/main/student.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Final Project Submission

Student Details:
* Student name: Lilian Muendo
* Student pace: part time
* Scheduled project review date/time: July 2024
* Instructor name: Samwel Jane and Veronica Isiaho
* Blog post URL: https://github.com/Lilian624/Microsoft-Movies-Project-1


**MOVIE STUDIO CREATION PROJECT**

**STEP 1: PROJECT OVERVIEW**

In this project we shall use data from already existing big movie companies exploring what types of films are currently doing the best at the box office and use our learnings to help and advise  Microsoft on how to create movie studios and on what type of films to create

**STEP 2: DESCRIPTION OF OUR DATASET**



IMPORTING LIBRARIES

Data Exploration and Cleaning - Load , explore and clean the provided data sets   i.e

imdb.title.basics

imdb.title.ratings

bom.movie_gross

Python Tools to Use


1.   Pandas for data manipulation and analysis
2.   NumPy for numerical data computations
3. Matplotlib for data visualization
4. Seaborn for data visualization






Data Importing , Exploration and Cleaning

In [4]:
# Import Libraries to be used to analyze data
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

In [6]:
#Loading the data sets
#load the first data set which is bom movie gross  in a csv version
movie_gross = pd.read_csv('/content/bom.movie_gross.csv')
#load the second data set which is title basics in a csv version
title_basics = pd.read_csv('/content/title.basics.csv')
#load the third data set which is title ratings in a csv version
title_ratings = pd.read_csv('/content/title.ratings.csv')

**Accessing Data in the Data Sets**

In [7]:
#Accessing information in the first data set (movie gross)
#this first code executes the number of rows and number of columns in the movie gross data
print(movie_gross.shape)
#this second code executes the summary of information contained in the rows and columns
print(movie_gross.info())

(3387, 7)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
 5   Unnamed: 5      0 non-null      float64
 6   Unnamed: 6      1 non-null      float64
dtypes: float64(3), int64(1), object(3)
memory usage: 185.4+ KB
None


In [10]:
#Print the first 5 rows of data in all 3 data sets
#print first 5 rows of Movie gross data
print("Movie Gross Data")
print(movie_gross.head(5))

#print first 5 rows of title basics data
print('Titl Basics Data')
print(title_basics.head(5))

#print first 5 rows of Title Ratings Data
print("Title Ratings Data")
print(title_ratings.head(5))

Movie Gross Data
                                         title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  Unnamed: 5  Unnamed: 6  
0     652000000  2010         NaN         NaN  
1     691300000  2010         NaN         NaN  
2     664300000  2010         NaN         4.0  
3     535700000  2010         NaN         NaN  
4     513900000  2010         NaN         NaN  
Titl Basics Data
      tconst                    primary_title              original_title  \
0  tt0063540                        Sunghursh                   Sunghursh   
1  tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
2  tt006

Displaying the summary statistics of the 3 data sets

Expectations


*   We will be able to generate the mean , minimun , standard deviation and count of our respective data sets
*   Addditionaly the maximun and minimun values which are also part of the execution will help us to make qualitative recommendations at the end



In [11]:
# Print summary statistics for each data set
print("Movie Gross Set")
print(movie_gross.describe())

print("Title Basics Data Set")
print(title_basics.describe())

print("Title Ratings Data Set")
print(title_ratings.describe())

Movie Gross Set
       domestic_gross         year  Unnamed: 5  Unnamed: 6
count    3.359000e+03  3387.000000         0.0         1.0
mean     2.874585e+07  2013.958075         NaN         4.0
std      6.698250e+07     2.478141         NaN         NaN
min      1.000000e+02  2010.000000         NaN         4.0
25%      1.200000e+05  2012.000000         NaN         4.0
50%      1.400000e+06  2014.000000         NaN         4.0
75%      2.790000e+07  2016.000000         NaN         4.0
max      9.367000e+08  2018.000000         NaN         4.0
Title Basics Data Set
          start_year  runtime_minutes
count  146144.000000    114405.000000
mean     2014.621798        86.187247
std         2.733583       166.360590
min      2010.000000         1.000000
25%      2012.000000        70.000000
50%      2015.000000        87.000000
75%      2017.000000        99.000000
max      2115.000000     51420.000000
Title Ratings Data Set
       averagerating      numvotes
count   73856.000000  7.385600e

Checking For Missing Values in the Data Sets

   This will help us identify if our rows and columns have all the data or there is missing values

   Afterwards we will do data cleaning.

In [14]:
# Check and print missing values in movie gross data set
print('Missing Values for Movie Gross Data')
print(movie_gross.isnull().sum())
# Check and print missing values in title_basics data set
print('Missing Values for Title Basics Data')
print(title_basics.isnull().sum())
# Check and print missing values in title ratings data set
print('Missing Values for Title Ratings Data')
print(title_ratings.isnull().sum())

Missing Values for Movie Gross Data
title                0
studio               5
domestic_gross      28
foreign_gross     1350
year                 0
Unnamed: 5        3387
Unnamed: 6        3386
dtype: int64
Missing Values for Title Basics Data
tconst                 0
primary_title          1
original_title        22
start_year             0
runtime_minutes    31739
genres              5408
dtype: int64
Missing Values for Title Ratings Data
tconst           0
averagerating    0
numvotes         0
dtype: int64


**Summary**


1.   For Movie gross Data: 5 missing in title, 28 missing in domestic gross, 1350 missing in foreign gross and row 3387 and 3386
2.   For Title Basics missing values: 1 in primary title, 22 in original title , 31739 in runtime minutes, 5408 in genre
3.   For Title Rating no missing values









Data Cleaning

Movie Gross Data

In [17]:
# Convert 'foreign_gross' to numeric, coerce errors to NaN
#This code changes foreign gross data from an object to nummeric
movie_gross['foreign_gross'] = pd.to_numeric(movie_gross['foreign_gross'], errors='coerce')

In [18]:
# Drop rows with missing 'studio' and 'domestic_gross' and assign to variable movie_gross_new
movie_gross_new = movie_gross.dropna(subset=['studio', 'domestic_gross'])

In [19]:
# Display cleaned data summary i.e data assigned to movie_gross_new
movie_gross_new.info()


<class 'pandas.core.frame.DataFrame'>
Index: 3356 entries, 0 to 3386
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3356 non-null   object 
 1   studio          3356 non-null   object 
 2   domestic_gross  3356 non-null   float64
 3   foreign_gross   2002 non-null   float64
 4   year            3356 non-null   int64  
 5   Unnamed: 5      0 non-null      float64
 6   Unnamed: 6      1 non-null      float64
dtypes: float64(4), int64(1), object(2)
memory usage: 209.8+ KB


In [20]:
#Check and Print First few rows and colums
print('Here are the first few rows and columns of our new cleaned Movie Gross Data')
movie_gross_new.head()

Here are the first few rows and columns of our new cleaned Movie Gross Data


title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

   foreign_gross  year  Unnamed: 5  Unnamed: 6  
0    652000000.0  2010         NaN         NaN  
1    691300000.0  2010         NaN         NaN  
2    664300000.0  2010         NaN         4.0  
3    535700000.0  2010         NaN         NaN  
4    513900000.0  2010         NaN         NaN

Data Cleaning

Title Basics Data

In [23]:
#Clean missing values in Title_Basics
#Start with: 'runtime_minutes' by filling with the median runtime value
#Checking for the median runtime
median_runtime = title_basics['runtime_minutes'].median()
#Replacing the missing runtime with the median runtime
title_basics['runtime_minutes'].fillna(median_runtime, inplace=True)

#Second filling in the missing 'genre' with 'Unknown' instead
title_basics['genres'].fillna('Unknown', inplace=True)


In [24]:
# Drop rows with missing 'original_title' and assign the result to variable name title_basics_new
title_basics_new = title_basics.dropna(subset=['original_title'])

In [25]:
# Display cleaned new data summary in title basics
title_basics_new.info()


<class 'pandas.core.frame.DataFrame'>
Index: 146122 entries, 0 to 146143
Data columns (total 6 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   tconst           146122 non-null  object 
 1   primary_title    146122 non-null  object 
 2   original_title   146122 non-null  object 
 3   start_year       146122 non-null  int64  
 4   runtime_minutes  146122 non-null  float64
 5   genres           146122 non-null  object 
dtypes: float64(1), int64(1), object(4)
memory usage: 7.8+ MB


In [26]:
#First few rows and columns

print('Here is the first rows and columns of the cleaned Title Basics Data')

title_basics_new.head()

Here is the first rows and columns of the cleaned Title Basics Data


tconst                    primary_title              original_title  \
0  tt0063540                        Sunghursh                   Sunghursh   
1  tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
2  tt0069049       The Other Side of the Wind  The Other Side of the Wind   
3  tt0069204                  Sabse Bada Sukh             Sabse Bada Sukh   
4  tt0100275         The Wandering Soap Opera       La Telenovela Errante   

   start_year  runtime_minutes                genres  
0        2013            175.0    Action,Crime,Drama  
1        2019            114.0       Biography,Drama  
2        2018            122.0                 Drama  
3        2018             87.0          Comedy,Drama  
4        2017             80.0  Comedy,Drama,Fantasy

Merging Title Basics & Title Ratings Data

Reason

These two data sets have a column in common which is the **tconst column** and this will help in analyzing  the data corresponding to the said column easily

In [27]:
#Goal : Is to Merge Title Basics and Ratings Datasets and assigning that to new variable name merged_title_data
merged_title_data = pd.merge(title_basics_new, title_ratings, on='tconst')

In [28]:
#Print and display the merged_title_data summary
merged_title_data.info()
merged_title_data.head()
merged_title_data.tail()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 73856 entries, 0 to 73855
Data columns (total 8 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   tconst           73856 non-null  object 
 1   primary_title    73856 non-null  object 
 2   original_title   73856 non-null  object 
 3   start_year       73856 non-null  int64  
 4   runtime_minutes  73856 non-null  float64
 5   genres           73856 non-null  object 
 6   averagerating    73856 non-null  float64
 7   numvotes         73856 non-null  int64  
dtypes: float64(2), int64(2), object(4)
memory usage: 4.5+ MB


tconst               primary_title              original_title  \
73851  tt9913084            Diabolik sono io            Diabolik sono io   
73852  tt9914286           Sokagin Çocuklari           Sokagin Çocuklari   
73853  tt9914642                   Albatross                   Albatross   
73854  tt9914942  La vida sense la Sara Amat  La vida sense la Sara Amat   
73855  tt9916160                  Drømmeland                  Drømmeland   

       start_year  runtime_minutes        genres  averagerating  numvotes  
73851        2019             75.0   Documentary            6.2         6  
73852        2019             98.0  Drama,Family            8.7       136  
73853        2017             87.0   Documentary            8.5         8  
73854        2019             87.0       Unknown            6.6         5  
73855        2019             72.0   Documentary            6.5        11